### PR TITLE
1525 add MultiScaleLoss

### LINKED ISSUE #1525: multiscale segmentation loss API
**Is your feature request related to a problem? Please describe.**
this ticket looks for a generic wrapper API to compute a given segmentation loss at multiple visual scales. the API should be compatible with the existing single-scale segmentation loss in monai

see also:
deepreg example: https://github.com/DeepRegNet/DeepReg/blob/a182f9d86fbd1ff9e392c511ff89403971be6454/deepreg/loss/label.py#L42-L47

### PR DESCRIPTION
**Pull request by kate-sann5100**
Signed-off-by: kate-sann5100 <yiwen.li@st-annes.ox.ac.uk>

Fixes #1525.

### Description
Add MultiScaleLoss

### Status
**Ready/Work in progress/Hold**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh --codeformat --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick`.
- [x] In-line docstrings updated.
- [x] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/losses/multi_scale.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+import torch
+from torch.nn.modules.loss import _Loss
+
+from monai.networks.layers import gaussian_1d, separable_filtering
+
+
+def make_gaussian_kernel(sigma: int) -> torch.Tensor:
+    if sigma <= 0:
+        raise ValueError(f"expecting postive sigma, got sign={sigma}")
```
Comment (wyli) [inline monai/losses/multi_scale.py:11]: some typos

File: monai/losses/multi_scale.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    kernel = gaussian_1d(sigma=torch.tensor(sigma), truncated=3, approx="sampled", normalize=False)
+    return kernel
+
+
+def make_cauchy_kernel(sigma: int) -> torch.Tensor:
+    if sigma <= 0:
+        raise ValueError(f"expecting postive sigma, got sign={sigma}")
```
Comment (wyli) [inline monai/losses/multi_scale.py:18]: typos in the message

File: monai/losses/multi_scale.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+                # no smoothing
+                losses.append(self.loss(y_pred, y_true))
+            else:
+                losses.append(
+                    self.loss(
+                        separable_filtering(y_pred, [self.kernel_fn(s)] * (y_true.ndim - 2)),
+                        separable_filtering(y_true, [self.kernel_fn(s)] * (y_true.ndim - 2)),
+                    )
+                )
+        loss = torch.mean(torch.stack(losses, dim=0), dim=0)
```
Comment (wyli) [inline monai/losses/multi_scale.py:77]: perhaps better to make this optional? like https://github.com/Project-MONAI/MONAI/blob/f7565bfcd207eb4cffff6d4a9e664c9a14e393fe/monai/losses/dice.py#L348-L355

**Review by wyli — COMMENTED** [review]
thanks! please see a few comments inline

**Review by wyli — APPROVED** [review]
thanks!
